cmd/kube-apiserver/app/server.go

```diff
@@ -115,6 +116,9 @@ func Run(s *options.ServerRunOptions) error {
 	if err = s.Authentication.Apply(genericConfig); err != nil {
 		return fmt.Errorf("failed to configure authentication: %s", err)
 	}
+	if err := preflight.RunAPIServerChecks(s); err != nil {
```

Comment:
Alright, now making it all the way to usage, I think I'd like `utilwait.PollImmediate(interval, duration, conditionfunc)` where you provide the condition func for testing etcd servers as a function on an struct.


```go
err := utilwait.PollImmediate(100 milliseconds, 10 seconds, NewEtcdCheck(serverlist).Condition)
if err != nil{
    return err
}
```

> It's not clear to me how this will make the resulting code any cleaner--or even work. WaitForAvailableEtcd is waiting on multiple go routines simultaneously, and signalling for all of them to stop when a verdict has been reached. Because utilwait.PollImmediate is polling one condition in one thread, I'm not sure what we would gain from it here. Could you please elaborate on how you picture it being used here?

Throughout the rest of the codebase (more or less), we use `wait.<something>` to wait on conditions.  Since the retry is done separately, there's no retry logic in your "check for etcd logic".  Since you're only trying once per attempt, there's no reason to have the stop channel (a bounded number of extra gofuncs just waiting to fail or succeed doesn't much matter), so your logic above gets to focus on "is an etcd up".